app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -295,7 +296,6 @@ def find_host(uuid = nil)
       if facet.nil?
         # check with candlepin if consumer is Gone, raises RestClient::Gone
         User.as_anonymous_admin { Resources::Candlepin::Consumer.get(uuid) }
-        fail HttpErrors::NotFound, _("Couldn't find consumer '%s'") % uuid
```

Comment:
This line was never executed as previous line would throw a `RestClient::Gone` error if the host does not exist.
In case if this line is reached when the host does exist, then the error should not be raised.